### PR TITLE
Catches possible exceptions and logs them

### PR DESCRIPTION
**Pull request by MarcosFleury**

### CODE REVIEW

File: Jobs.Fetcher.Twitter/Jobs/Ads/AdsAccountsQuery.cs [hunk 1]
```diff
@@ -40,10 +40,14 @@ public class AdsAccountsQuery : AbstractTwitterFetcher {
...
-                while (!iterator.Completed) {
-                    var adsAccountPage = iterator.NextPageAsync().GetAwaiter().GetResult();
-                    DbWriter.WriteAdsAccounts(user.Id, username, adsAccountPage.Content, adsContext, GetLogger());
+                try {
+                    while (!iterator.Completed) {
+                        var adsAccountPage = iterator.NextPageAsync().GetAwaiter().GetResult();
+                        DbWriter.WriteAdsAccounts(user.Id, username, adsAccountPage.Content, adsContext, GetLogger());
+                    }
+                }catch (Exception e) {
+                    GetLogger().Error($"Could not fetch Twitter Video Libraries for {username}");
```
Comment (moraesvic) [inline Jobs.Fetcher.Twitter/Jobs/Ads/AdsAccountsQuery.cs:49]: I might be getting something wrong, but do you need GetLogger here? I always assumed the existence of `Logger` in the `AbstractJob` class. If this is correct, you could also change it for the function call `DbWriter.WriteAdsAccounts` above

File: Jobs.Fetcher.Twitter/Jobs/Ads/CampaignsQuery.cs [hunk 2]
```diff
@@ -37,9 +37,14 @@ public class CampaignsQuery : AbstractTwitterFetcher {
             var adsAccountIds = DbReader.GetAdsAccountIds(username, dbContext);
 
             void ProccessCampaignResult(string accountId, ITwitterRequestIterator<CampaignsResponse, string> iterator) {
```
Comment (moraesvic) [inline Jobs.Fetcher.Twitter/Jobs/Ads/CampaignsQuery.cs:39]: Typo: Proccess -> Process

File: Jobs.Fetcher.Twitter/Jobs/Ads/CampaignsQuery.cs [hunk 3]
```diff
@@ -37,9 +37,14 @@ public class CampaignsQuery : AbstractTwitterFetcher {
...
-                while (!iterator.Completed) {
-                    var campaignsPage = iterator.NextPageAsync().GetAwaiter().GetResult();
-                    DbWriter.WriteCampaigns(accountId, campaignsPage.Content, dbContext, GetLogger());
+                try {
+                    while (!iterator.Completed) {
+                        var campaignsPage = iterator.NextPageAsync().GetAwaiter().GetResult();
+                        DbWriter.WriteCampaigns(accountId, campaignsPage.Content, dbContext, GetLogger());
+                    }
+                }catch (Exception e) {
```
Comment (moraesvic) [inline Jobs.Fetcher.Twitter/Jobs/Ads/CampaignsQuery.cs:45]: A suggestion for future improvement: separating this between errors that could arise from external library `FlycatcherAds` and errors which are "our fault", e.g., database integrity check fails

Comment (MarcosFleury) [reply Jobs.Fetcher.Twitter/Jobs/Ads/CampaignsQuery.cs:45]: FlycatcherAds has a side that is ours and another that is external. But you are right, it is a good idea

**Review by moraesvic — APPROVED** [review]

**Pull request merged**